core/domain/exp_domain.py

```diff
@@ -1145,8 +1138,17 @@ def update_interaction_answer_groups(self, answer_groups_list):
                         'Expected rule_inputs to be a dict, received %s'
                         % rule_inputs)
                 for param_name, value in rule_inputs.iteritems():
-                    param_type = rule_domain.get_obj_type_for_param_name(
-                        matched_rule, param_name)
+                    param_list = rule_domain.get_param_list(
+                        matched_rule_description)
+
+                    param_type = None
```

Comment:
Since ``get_param_list()`` is only used in rule_domain_test and exp_domain and rule_domain* will probably be removed, we might as well just implement a ``get_param_type_from_rule_description()`` function which combines this logic with the existing logic in ``get_param_list()``. It keeps this code cleaner and avoids having a strange function which we only partially using.